packages/dds/sequence/src/test/overlappingIntervalsIndex.spec.ts

```diff
@@ -0,0 +1,460 @@
+
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 2, end: 3 },
+				{ start: 5, end: 6 },
+			]);
+		});
+
+		it("when existing interval across multiple segments", () => {
+			// Add intervals which are across more than one segments
+			const interval4 = collection.add(1, 3, IntervalType.SlideOnRemove);
+			const interval5 = collection.add(2, 5, IntervalType.SlideOnRemove);
+			const interval6 = collection.add(1, 6, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+			overlappingIntervalsIndex.add(interval5);
+			overlappingIntervalsIndex.add(interval6);
+
+			results = queryIntervalsByPositions(0, 1);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 1, end: 3 },
+				{ start: 1, end: 6 },
+			]);
+
+			results = queryIntervalsByPositions(1, 3);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 1, end: 3 },
+				{ start: 1, end: 6 },
+				{ start: 2, end: 3 },
+				{ start: 2, end: 5 },
+			]);
+
+			results = queryIntervalsByPositions(5, 5);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 6 },
+				{ start: 2, end: 5 },
+				{ start: 5, end: 6 },
+			]);
+		});
+
+		it("when adding duplicate intervals to the index", () => {
+			const interval4 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			const interval5 = collection.add(2, 3, IntervalType.SlideOnRemove);
+			const interval6 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+			overlappingIntervalsIndex.add(interval5);
+			overlappingIntervalsIndex.add(interval6);
+
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 1, end: 1 },
+				{ start: 1, end: 1 },
+				{ start: 2, end: 3 },
+				{ start: 2, end: 3 },
+				{ start: 5, end: 6 },
+			]);
+		});
+
+		it("when removing intervals from the index", () => {
+			overlappingIntervalsIndex.remove(interval2);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 5, end: 6 },
+			]);
+
+			// Add and remove duplicate intervals
+			const interval4 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+			overlappingIntervalsIndex.remove(interval1);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 5, end: 6 },
+			]);
+
+			overlappingIntervalsIndex.remove(interval4);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [{ start: 5, end: 6 }]);
+		});
+
+		it("when inserting or appending additional segments to the string", () => {
```

Comment:
FWIW, I'm not sure these test cases provide much coverage we don't get elsewhere. But the diagrams are pretty nice and I'd believe when they do fail, they're easier to debug than the average other test that also fails. so i'm fine with keeping them.

Reply:
no action in the code needed here. I'm curious to hear your thought process on why you decided to write this test case (and subsequent one).

Reply:
In cases where we only update the text (segments) of the shared string, we don't explicitly modify the intervals stored in the index or update the interval collection directly. However, I look forward to seeing if the intervals in the index will automatically sync with the changes made to the shared string. That's why I'm interested in adding these tests.